### DISCUSSION TITLE
API daily limit

### DISCUSSION THREAD

**Discussion by LittleBigDev** [cytech/Home-Assistant-wundergroundpws#78] [General]
Hello,

I read that everything is updated every 5 minutes. Does that mean it makes 24 * 60 / 5 = 288 calls per day ?  
If yes, then why does my data starts being updated every day at 00:15 am (UTC) and stops at 06:10 pm (UTC) ?  
No data between 06:10 pm and 00:15 am the next day.

In this time range, there are 215 refreshes.  
What I can't understand is **the stats from wunderground.com**. Every day, there are approximately 3000 calls (1500 successes, 1500 failures). That would mean 14 calls per refresh. Every 5 minutes.  
I was not able to find the calls daily limit but it looks like it's around 3000... 

How is that possible ? Did I forget to check something ? Config problem ? Anything else ?

**Comment by rpitera** [comment]
I have no setting for the refresh, so I'm using the defaults between the component code and HA.  I just checked and I am seeing an average of 500-570 successful hits a day.  I don't think there even is a way to override the refresh rate.

Honestly though, I use WeeWX with my Acurite 5in1 and get real-time data via MQTT, so I only use the component for weather predictions and precip rates for notifications, so 5 minutes is fine for me.

**Comment by LittleBigDev** [comment]
Ok so obviously my API key is being used by something else. I know I used it for a self-hosted meteo application. Maybe this is it. 
I regenerated my WU API key and updated it only in HA. We'll see...

**Comment by rpitera** [comment]
As unlikely as it may sound, that did happen to me a long time ago before WU was purchased.  I had posted some code somewhere and forgot to sanitize the API and believe it or not someone used it.  Could be what's happening here.  EIther way you are smart to regen the API key.

**Comment by LittleBigDev** [comment]
Looks like I was right. Now I'm around 500 successes per day, and 0 failures.

I will delete this sh*tty meteo app and go on with HA only :)

**Comment by cytech** [comment]
closed as resolved

**Comment by cytech** [comment]
closed as resolved
